src/lib/es2019.array.d.ts

```diff
@@ -1,3 +1,10 @@
+type Flat<A, D extends number> = {

```

Comment:
we should check for existing types named `Flat` in some code base -- probably it's fine, but might not be.

Reply:
Probably Definitely Typed is a big enough sample -- grepping for `class Flat`, `interface Flat`, `type Flat` should give you (almost?) all of the uses.

Reply:
* [seen](http://www.rice-boy.com/see/index.php?c=189) has `class Flat` (appropriate for a 3D renderer)
* array.prototype.flat has `interface Flat` (and `FlatMap`)

Seen is barely used, but array.prototype.flat usage is quite widespread.